### DISCUSSION TITLE
what 'problems' does prest solve?

### DISCUSSION THREAD

**Discussion by parsonsmatt** [prest/prest#41] [API Server] [answered]
The README states:

> There is the PostgREST written in haskell, keep a haskell software in production is not easy job, with this need that was born the pREST.

I would (and am sure many others would also) love to hear more about the issues you had. In the same vein of Pusher's [articles](https://blog.pusher.com/latency-working-set-ghc-gc-pick-two/) on switching [from Haskell to Go](https://blog.pusher.com/golangs-real-time-gc-in-theory-and-practice/), would you be willing to elaborate on the issues you encountered?

**Comment by avelino** [comment]
Today the postgrest is not distributed a binary, need to install Haskell to compile. pREST in its 1.0 version will be binary distributed (cross compiling, run in Windows by example), so you don't need to have anything installed on the server to run.

How old is Haskell and how many developers exist? Go has less years and more developers, if you need to fix a problem in postgrest turn where?

**Comment by parsonsmatt** [comment]
I was about to go raise distributed binaries as an issue on the PostgREST repo, but saw [that they do distribute binaries](https://github.com/begriffs/postgrest/releases/tag/v0.3.2.0) for a number of platforms.

The point about Haskell's developer situation is a good point. As a Haskell dev, I'd just fix it myself -- and I recognize that there are fewer Haskell devs than Go devs.

Thanks for the input! :smile:

**Comment by avelino** [comment]
The reason the pREST was born was because of a lack of developer, I am using pREST in my company, I know haskell and my developers nothing, if you have any problem (in production) just sits in my hand (hand want that).

It was great to your questioning.

**Comment by codeyash** [comment]
I personally don't like the idea of using go lang.
Ruby, Python or PHP makes more sense to me. But hey it's me.

**Comment by avelino** [comment]
@codeyash pREST is software distributed as binary, you do not need to program in Go lang to use

**Comment by xpufx** [comment]
Maybe it's time to remove that blurb and actually provide info about the strengths of this project. (postgrest is distributed as a single binary now. Even if it weren't you could dockerize it. Since it sits in the middle its resource needs aren't extensive. It doesn't even need persistent storage)

FYI that blurb turned me off and I went and installed the docker for postgrest itself even though I program in go and not haskell.

**Comment by avelino** [comment, marked as answer]
Some strengths of pREST:
- **SQL script Execution** (endpoint based on predefined SQL) - resource was born by being software engineer not DBA, have no version control of functions (PostgreSQL resource) does not like me, we can version the script that creates the function, but when running it in PostgreSQL we have to `drop function`: https://docs.postgres.rest/executing-sql-scripts/
- **Extending (using with framework)** - it is possible to use pREST as a Go framework, if you want to create different endpoints (that do other work besides talking to the database) it is possible and you get all the existing endpoints in pREST without having to implement: https://docs.postgres.rest/prest-as-web-framework/ *
- **Plugins**: it is possible to write plugins and extend ever further your application.

* Framework was deprecated but is still available to use

**Comment by newmizanur** [comment]
PostgREST doesn't have easy support for multiple database, I believe.

**Reply by avelino** [reply]
Not yet accepted, we have a discussion about it here, your contribution will be very welcome

#435
